FreeRTOS-Plus/Test/CBMC/proofs/UDP/vProcessGeneratedUDPPacket/vProcessGeneratedUDPPacket_harness.c

```diff
@@ -0,0 +1,92 @@
+/* Standard includes. */
+#include <stdint.h>
+#include <stdio.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "queue.h"
+#include "list.h"
+#include "semphr.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+
+#include "FreeRTOS_ARP.h"
+#include "FreeRTOS_UDP_IP.h"
+//#include "FreeRTOS_TCP_IP.h"
+#include "FreeRTOS_IP_Private.h"
+
+
+
+uint16_t usGenerateChecksum( uint16_t usSum, const uint8_t * pucNextData, size_t uxByteCount ) 
+{
+	uint16_t usChecksum;
+
+	/* Return any random value of checksum since it does not matter for CBMC checks. */
+	return usChecksum;
+}
+
+uint16_t usGenerateProtocolChecksum( const uint8_t * const pucEthernetBuffer, size_t uxBufferLength, BaseType_t xOutgoingPacket )
+{
+	uint16_t usProtocolChecksum;
+
+	/* Return random value of checksum since it does not matter for CBMC checks. */
+	return usProtocolChecksum;
+}
+
+/* This function has been tested separately. Therefore, we assume that the implementation is correct. */
+void vARPRefreshCacheEntry( const MACAddress_t * pxMACAddress, const uint32_t ulIPAddress ) { }
```

Comment:
As in other pull requests, anything to check?  Null?  Assumptions made in the memory safety proofs of RefreshCacheEntry and GenerateRequestPacket?

Reply:
I have re-worked the proof. Thanks.